### ISSUE TITLE
iOS ActionBarItems overlapping before page load 

### ISSUE THREAD

**Issue by firescript** [NativeScript/NativeScript#2179]
hey guys I have an action bar thats labels overlap on load then pop to the right spots when its loaded. 

tns version: 2.0.0
package.json: "version": "2.0.0",

Here is a barebones project showing the issue: 
[actionBarGlitch.zip](https://github.com/NativeScript/NativeScript/files/282380/actionBarGlitch.zip)

Here is a video showing the issue.
[teamreps nav bug.zip](https://github.com/NativeScript/NativeScript/files/282215/teamreps.nav.bug.zip)

markup looks like this:

```
    <Page.actionBar>
        <ActionBar>
            <ActionBar.titleView>
                <Label text="TEAMREPS" class="logo" />
            </ActionBar.titleView>
            <ActionBar.actionItems>                
                <ActionItem ios.position="left" tap="logout">
                    <ActionItem.actionView>

                            <Label text="&#xf013;" class="font-awesome-icon nav-icon"  />

                    </ActionItem.actionView>
                </ActionItem>                
                <ActionItem ios.position="right" tap="addTeam">
                    <ActionItem.actionView>

                            <Label text="&#xf0fe;" class="font-awesome-icon nav-icon"  />

                    </ActionItem.actionView>
                </ActionItem>
            </ActionBar.actionItems>            
        </ActionBar>
    </Page.actionBar> 
```

it also does it with in another view with this action bar:

```
    <Page.actionBar>
        <ActionBar>
            <ActionBar.titleView>
                <Label text="{{ pageTitle }}" />
            </ActionBar.titleView>
            <ActionBar.actionItems>
                <ActionItem ios.position="left" tap="dashboard">
                    <ActionItem.actionView>
                        <StackLayout>
                            <Label text="&#xf060;" class="font-awesome-icon nav-icon"  />
                        </StackLayout>
                    </ActionItem.actionView>
                </ActionItem>  
            </ActionBar.actionItems>
        </ActionBar>
    </Page.actionBar>
```

**Comment by NickIliev** [comment]
Looks like ActionItem ios.position="right is reset on navigation transition.. possibly related to [https://github.com/NativeScript/NativeScript/issues/2171](https://github.com/NativeScript/NativeScript/issues/2171)

**Comment by firescript** [comment]
Alright it isn't related to the ios.position, I just checked. It seems to be related to any use case of font- in the css, so font-family, font-size, etc. Below is a bare bones project showing the exact issue on a new project.

[actionBarGlitch.zip](https://github.com/NativeScript/NativeScript/files/282378/actionBarGlitch.zip)

**Comment by sitefinitysteve** [comment]
I can vouch for seeing this in his app, and replicating in the hello-world... second we add any font-family\size css to anything in the actionbar there... popping

**Comment by sitefinitysteve** [comment]
So @NickIliev it seems this totally breaks ANDROID as well... I've been killing myself for the last 4 hours trying to track down why this app just fails to load on android.  Turns out it's the exact same issue... in that adding css in the custom actionbar breaks things.  On iOS its just jumpy buttons, android outright tanks.

Video:
http://www.screencast.com/t/NYFBuORVUpo

Sidenote: Css in question
```

**Comment by NickIliev** [comment]
@sitefinitysteve
 I have played with this code and what seems to be the problem (in Android) is the css-property **_margin_** - when you set margin (or some of the sub-properties **_margin-top, margin-bottom, margin-left, margin-right_**) on element (in this case Label) which is part of **_ActionBar.titleView_**

As a workaround I have tried to wrap the label in container element like this:  
```
<StackLayout >
    <Label text="TITLE" class="logo" />
</StackLayout>
```

This will solve your Label styling problem but however the main problem persist - now we can't set certain CSS-properties on the StackLayout.(same error thrown on Android with **margin**) - I will log this behaviour in new issue as this one has different focus.

@firescript 

Indeed when you use the long syntax for creating ActionItem the described behaviour is observed in iOS - on navigation the position of the ActionItems is not rendered correctly.As a temporary workaround uou can use the short syntax but then you can set fontIcons and must use the predefined system icons for iOS and Android.
```
<ActionBar.actionItems>
    <ActionItem  ios.systemIcon="12" android.systemIcon="ic_menu_search" tap="otherpage" />
    <ActionItem  ios.systemIcon="16" android.systemIcon="ic_menu_delete" tap="otherpage"  />
</ActionBar.actionItems>    
```

**Comment by NickIliev** [comment]
To reproduce the issue create two pages with ActionBarItems using the extended syntax.
When navigation from pageA to pageB the ActionBarItems are placed in the top-left corner (iOS) at the beggining and then on final load the actual positions are applied. 

**pageA.xml** & pageB.xml
```
<Page loaded="onNavigatingTo" xmlns:lv="nativescript-telerik-ui/listview" xmlns="http://www.nativescript.org/tns.xsd">
    <Page.actionBar>
        <ActionBar>
            <ActionBar.titleView>
                <StackLayout>
                    <Label text="TITLE" />
                </StackLayout>
            </ActionBar.titleView>
        <ActionBar.actionItems>                
            <ActionItem tap="onTap">
                <ActionItem.actionView>
                    <StackLayout>
                        <Label text="Item 1"  />
                    </StackLayout>
                </ActionItem.actionView>
            </ActionItem>                
            <ActionItem  tap="onTap">
                <ActionItem.actionView>
                    <StackLayout>
                        <Label text="Item 2"  />
                    </StackLayout>
                </ActionItem.actionView>
            </ActionItem>
        </ActionBar.actionItems>              
      </ActionBar>
    </Page.actionBar>         
</Page>
```

pageA.ts (pageB.ts identical code only change moduleName value)
```
import { EventData } from "data/observable";
import { Page } from "ui/page";
import frameModule = require("ui/frame");

export function navigatingTo(args: EventData) {
    var page = <Page>args.object;
}

export function onTap (args: EventData){
     frameModule.topmost().navigate({
        moduleName: "main-page",
        clearHistory: true
    });    
};
```

**Comment by tskweres** [comment]
I can confirm when using icon fonts in the action bar, this is a problem still

**Comment by tskweres** [comment]
If you download and run this repo, it demonstrates how the actionbar with font icons inside will weirdly 'snap' into place if you use ios.position:

https://github.com/tskweres/tinder-clone

**Comment by tzraikov** [comment]
Will be addressed with #3093
